packages/lodestar/src/tasks/tasks/pruneBlockArchive.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @module tasks
+ */
+
+import {computeStartSlotAtEpoch} from "@chainsafe/lodestar-beacon-state-transition/src/util/epoch";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {GENESIS_EPOCH} from "@chainsafe/lodestar-params";
+import {ILogger} from "@chainsafe/lodestar-utils";
+import {IBeaconClock} from "../../chain";
+import {getMinEpochForBlockRequests} from "../../constants";
+import {IBeaconDb} from "../../db";
+import {ITask} from "../interface";
+export interface IPruneBlockArchiveModules {
+  db: IBeaconDb;
+  clock: IBeaconClock;
+  logger: ILogger;
+}
+
+/**
+ * Since we only need to serve MIN_EPOCHS_FOR_BLOCK_REQUESTS of history blocks.
+ * We can delete it periodically.
+ */
+export class PruneBlockArchiveTask implements ITask {
```

Comment:
For testnets we may want to keep all blocks for debugging purposes. Should we add a flag to prevent nuking the block db for everyone?

Reply:
yeah I think we should add flag, not just for testnet but for other consumers which might need to store all blocks